OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbySavedSettingsBrowserLogic.cs

```diff
@@ -0,0 +1,351 @@
+					if (isSavePanel)
+					{
+						item.IsSelected = () => saveTextField.Text == Path.GetFileNameWithoutExtension(item.ItemKey);
+						item.OnDoubleClick = Save;
+					}
+					else
+					{
+						item.IsSelected = () => selectedSave == item.ItemKey;
+						item.OnDoubleClick = Load;
+					}
+
+					var title = Path.GetFileNameWithoutExtension(savePath);
+					var label = item.Get<LabelWithTooltipWidget>("TITLE");
+					WidgetUtils.TruncateLabelToTooltip(label, title);
+
+					var date = File.GetLastWriteTime(savePath).ToString("yyyy-MM-dd HH:mm:ss");
+					item.Get<LabelWidget>("DATE").GetText = () => date;
+
+					saveList.AddChild(item);
+				}
+			}
+		}
+
+		void Save()
+		{
+			var globalSettings = orderManager.LobbyInfo.GlobalSettings.Serialize();
+			var options = globalSettings.Value.ToDictionary()["Options"];
+			var unlockedOptions = options.Nodes.FindAll(optionYaml => optionYaml.Value.ToDictionary()["IsLocked"].Value == "False");
+			var optionsYaml = new List<MiniYamlNode>()
+			{
+				new MiniYamlNode(SettingsKeyMapId, map.Uid),
+				new MiniYamlNode(SettingsKeyOptions, new MiniYaml(options.Value, unlockedOptions))
+			};
+
+			if (!Directory.Exists(baseSavePath))
+				Directory.CreateDirectory(baseSavePath);
+
+			var filename = SettingsFilePath(saveTextField.Text);
+
+			Action inner = () =>
+			{
+				optionsYaml.WriteToFile(filename);
+				Ui.CloseWindow();
+			};
+
+			if (File.Exists(filename))
+			{
+				ConfirmationDialogs.ButtonPrompt(
+					title: "Overwrite saved settings?",
+					text: "Overwrite {0}?".F(saveTextField.Text),
+					onConfirm: inner,
+					confirmText: "Overwrite",
+					onCancel: () => { });
+			}
+			else
+				inner();
+		}
+
+		void Load()
+		{
+			var optionsYaml = MiniYaml.FromFile(selectedSave);
+
+			Ui.CloseWindow();
+
+			var savedMap = optionsYaml.Find(node => node.Key == SettingsKeyMapId).Value.Value;
+			if (savedMap != map.Uid)
+				orderManager.IssueOrder(Order.Command("map {0}".F(savedMap)));
+
+			var savedOptions = optionsYaml.Find(node => node.Key == SettingsKeyOptions);
+			foreach (var option in savedOptions.Value.Nodes)
+			{
+				string optionName = option.Key;
+				string optionValue = option.Value.ToDictionary()["Value"].Value;
+				orderManager.IssueOrder(Order.Command("option {0} {1}".F(optionName, optionValue)));
+			}
+		}
+
+		void Rename(string oldName, string newName)
+		{
+			try
+			{
+				var oldPath = SettingsFilePath(oldName);
+				var newPath = SettingsFilePath(newName);
+				File.Move(oldPath, newPath);
+
+				saves[saves.IndexOf(oldPath)] = newPath;
+				foreach (var c in saveList.Children)
+				{
+					var item = c as ScrollItemWidget;
+					if (item == null || item.ItemKey != oldPath)
+						continue;
+
+					item.ItemKey = newPath;
+					item.Get<LabelWidget>("TITLE").GetText = () => newName;
+				}
+
+				if (selectedSave == oldPath)
+					selectedSave = newPath;
+			}
+			catch (Exception ex)
+			{
+				Console.WriteLine(ex);
```

Comment:
Should we use `Game.Debug` here so the user sees there was some sort of error?

Reply:
Sounds good to me. Tested this and the error appears in the lobby chat window. The user doesn't see this _immediately_ because they are currently in the settings save browser but they do see it when they navigate back.